From 48ad5c23680c81614663e09c6586ebeb26bf8c18 Mon Sep 17 00:00:00 2001
From: Richard Marko <srk@48.io>
Date: Mon, 16 Oct 2023 15:26:33 +0200
Subject: [PATCH] util/kconfig: chmod +w before savedefconfig

This prevents a headscratcher when .config in root doesn't have a write
permission bit set which causes a build failure of savedefconfig
not able to write to copied file, for example

*** Error while saving defconfig to:
  build/mainboard/emulation/qemu-i440fx/cbfs-file.eU5E0t.out.tmp2

Change-Id: I2e7d35c9f6e8add3e7438d163850bc5fda5a99b2
Signed-off-by: Richard Marko <srk@48.io>
---
 util/kconfig/Makefile.inc | 1 +
 1 file changed, 1 insertion(+)

Index: kconfig/Makefile.inc
===================================================================
--- kconfig.orig/Makefile.inc
+++ kconfig/Makefile.inc
@@ -34,6 +34,7 @@ oldconfig: KCONFIG_STRICT=
 
 savedefconfig: $(objk)/conf
 	cp $(DOTCONFIG) $(DEFCONFIG)
+	chmod +w $(DEFCONFIG)
 	$< --savedefconfig=$(DEFCONFIG) $(KBUILD_KCONFIG)
 
 FORCE:
